source/common/http/conn_manager_impl.cc

```diff
@@ -1141,7 +1167,15 @@ void ConnectionManagerImpl::ActiveStream::encodeHeaders(ActiveStreamEncoderFilte
   resetIdleTimer();
   disarmRequestTimeout();
 
-  std::list<ActiveStreamEncoderFilterPtr>::iterator entry = commonEncodePrefix(filter, end_stream);
+  // Only do base state setting on the initial call. Subsequent calls for filtering do not touch
```

Comment:
Curious, would it work to retain commonEncodePrefix with an assert that if filter is set, that *(filter->entry()))->iterate_from_current_filter_ is false?

I'd think that couldn't be set until after we decode headers but I could easily be missing something

Reply:
Good idea! This works without the latest change. But in the latest change, I reset iteration_state_ at the beginning of commonContinue, iterate_from_current_filter_ may be true for doHeaders(). I introduced a bool always_start_next in commonEncodePrefix() to indicate if we should check iterate_from_current_filter_ in commonEncodePrefix(). I also created a similar function for decoders: commonDecodePrefix(). I am open to ideas that can make it better/cleaner.

Reply:
But how can we be stopped and still need to call doHeaders? We have to call doHeaders and get a StopAllIIteration to set iterate_from_current_filter_ true for a given filter.  At that point headers_continued_ is true and we won't doHeaders again, so how can we be back under commonEncodePrefix for header processing with  iterate_from_current_filter_ true?